### ISSUE TITLE
add output to stream of lines (seq string) ?

### ISSUE THREAD

**Issue by jkone27** [CaptnCodr/Fli#63]
would be useful to process the output as stream, like less or more commands, so as a stream of lines, and to have control to interrupt or modify the behaviour of code based on content

e.g.

`dotnet test` |> break if containing 'Exception' so it will stop at the first error in tests instead of running the full suite

**Comment by CaptnCodr** [comment]
Hey @jkone27, thank you for raising an issue. 
That's a good idea. I'll tackle this soon-ish.

**Comment by jkone27** [comment]
yes i think this could be useful to chain the printing of executing command while a command is still running or is not exiting, 
maybe also in this contrext would be useful to add a way to chain a command only when the previous one was "killed" with CTRL+C, so that instead of exiting the program, it will bring back the execution to the next step first


```
cli { ... some long running process printing } 
|> some cleanup tasks to run anyways even if the above program is stopped
```

one example case is when running python scripts you might want to exit the virtualenvironment always at the end, so i need to run a "closing" python command even if the running program is interrupted... but this is a very special case i guess.

**Comment by CaptnCodr** [comment]
Is this the wrong issue for your comment? 😄

**Comment by CaptnCodr** [comment]
Anyways, I think that is a special case, yes.
My first thought is to await for this process and look at the exitCode of the process and run special this for cleanup.

**Comment by CaptnCodr** [comment]
Thoughts on this:
`Output.Stream of standardOutput: Stream * standardError: Stream`
Filling `standardOutput` from `Process.StandardOutput` (`stdout`) & `standardError` from `Process.StandardError` (`stderr`)

**Comment by johnW-ret** [comment]
Bump.

This is a great wrapper but there's a bigger issue other than just having eager access to shell output. Upon [inspection of source](https://github.com/CaptnCodr/Fli/blob/334b1bc12456bae53cc59db22dd23ed9434343a9/src/Fli/Domain.fs#L120C1-L121C1), it appears that output text is always captured in memory - [even when writing to a file](https://github.com/CaptnCodr/Fli/blob/334b1bc12456bae53cc59db22dd23ed9434343a9/src/Fli/Command.fs#L199). It appears that the only cases where `Option.Text` can be `None` is when shell output is empty (?).

This breaks all use cases where a script prints out more text than there is memory available, and the library will capture output as memory even if it is not used by the consuming program.

A good start would be to rewrite the `File` option to simply forward writes to a file realtime and flush at the end but I'm not sure how changes could be made without introducting breaking changes as changing `Output.Text` to `None` in this case would technically not be correct. This, of course, would not fix the issue where output is stored in memory even when not used.

**Comment by CaptnCodr** [comment]
@johnW-ret Does PR #80 meet those requirements you mentioned?
